src/SprykerEco/Zed/Unzer/Business/Payment/Processor/Charge/UnzerCreditCardChargeProcessor.php

```diff
@@ -185,77 +182,149 @@ protected function createUnzerCharge(
      * @param \Generated\Shared\Transfer\UnzerChargeTransfer $unzerChargeTransfer
      * @param \Generated\Shared\Transfer\OrderTransfer $orderTransfer
      * @param \Generated\Shared\Transfer\PaymentUnzerOrderItemCollectionTransfer $paymentUnzerOrderItemCollectionTransfer
+     * @param array<int> $salesOrderItemIds
      *
      * @return \Generated\Shared\Transfer\UnzerChargeTransfer
      */
     protected function addExpensesToUnzerChargeTransfer(
         UnzerChargeTransfer $unzerChargeTransfer,
         OrderTransfer $orderTransfer,
-        PaymentUnzerOrderItemCollectionTransfer $paymentUnzerOrderItemCollectionTransfer
+        PaymentUnzerOrderItemCollectionTransfer $paymentUnzerOrderItemCollectionTransfer,
+        array $salesOrderItemIds
     ): UnzerChargeTransfer {
-        if (
-            $orderTransfer->getExpenses()->count() === 0
-            || $this->getChargedItemsCount($paymentUnzerOrderItemCollectionTransfer) > 0
-        ) {
+        if ($orderTransfer->getExpenses()->count() === 0) {
             return $unzerChargeTransfer;
         }
 
         $expensesAmount = 0;
         foreach ($orderTransfer->getExpenses() as $expenseTransfer) {
-            $expensesAmount += $this->getExpensesAmountForOrderExpense($orderTransfer, $expenseTransfer);
+            if (!$this->expenseHasRelatedChargeItems($expenseTransfer, $orderTransfer, $salesOrderItemIds)) {
+                continue;
+            }
+
+            if ($this->expenseHasAlreadyChargedItems($expenseTransfer, $orderTransfer, $paymentUnzerOrderItemCollectionTransfer)) {
+                continue;
+            }
+
+            $expensesAmount += $expenseTransfer->getSumPriceToPayAggregationOrFail();
+            $unzerChargeTransfer->addChargedSalesShipmentId($expenseTransfer->getShipmentOrFail()->getIdSalesShipmentOrFail());
         }
 
         return $unzerChargeTransfer->setAmount($unzerChargeTransfer->getAmount() + $expensesAmount);
     }
 
+    /**
+     * @param \Generated\Shared\Transfer\PaymentUnzerOrderItemTransfer $paymentUnzerOrderItemTransfer
+     *
+     * @return bool
+     */
+    protected function isPaymentUnzerOrderItemAlreadyCharged(PaymentUnzerOrderItemTransfer $paymentUnzerOrderItemTransfer): bool
+    {
+        return $paymentUnzerOrderItemTransfer->getStatus() === UnzerConstants::OMS_STATUS_PAYMENT_COMPLETED
+            || $paymentUnzerOrderItemTransfer->getStatus() === UnzerConstants::OMS_STATUS_CHARGE_REFUNDED;
+    }
```

Comment:
It's just a matter of taste, however 
```suggestion
        return in_array($paymentUnzerOrderItemTransfer->getStatus(), [
            UnzerConstants::OMS_STATUS_PAYMENT_COMPLETED,
            UnzerConstants::OMS_STATUS_CHARGE_REFUNDED,
        ], true);
```